### PR TITLE
 feat(azure-cli) install Azure CLI + autoupdate 

### PR DESCRIPTION
**Pull request by dduportal**
As part of https://github.com/jenkins-infra/helpdesk/issues/2789, it appears that the `az` Azure CLI is required (ref. https://github.com/jenkins-infra/pipeline-library/pull/348).

Only the `updatecli diff` step should fail

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -108,5 +116,6 @@ LABEL io.jenkins-infra.tools.aws-iam-authenticator.version="latest"
 LABEL io.jenkins-infra.tools.jenkins-agent.version="${JENKINS_AGENT_VERSION}"
 LABEL io.jenkins-infra.tools.doctl.version="${DOCTL_VERSION}"
 LABEL io.jenkins-infra.tools.ruby.version="${RUBY_VERSION}"
+LABEL io.jenkins-infra.tools.azure-cli.version="${AZ_CLI_VERSION}"
```
Comment (lemeurherve) [inline Dockerfile:119]: `azure-cli` should be added at the end of line 103

File: cst.yml [hunk 2]
```diff
@@ -5,7 +5,7 @@ metadataTest:
       value: "/home/helm/.helm"
   labels:
     - key: io.jenkins-infra.tools
-      value: "helm,kubectl,helmfile,sops,aws-cli,aws-iam-authenticator,yamllint,updatecli,jenkins-agent,doctl,jq,yq,ruby"
+      value: "helm,kubectl,helmfile,sops,aws-cli,aws-iam-authenticator,azure-cli,yamllint,updatecli,jenkins-agent,doctl,jq,yq,ruby"
```
Comment (lemeurherve) [inline cst.yml:8]:
```suggestion
      value: "helm,kubectl,helmfile,sops,aws-cli,aws-iam-authenticator,yamllint,updatecli,jenkins-agent,doctl,jq,yq,ruby,azure-cli"
```
See previous comment on Dockerfile

File: updatecli/updatecli.d/azure-cli.yml [hunk 3]
```diff
@@ -0,0 +1,80 @@
+---
+title: "Bump azcli version"
```
Comment (lemeurherve) [inline updatecli/updatecli.d/azure-cli.yml:2]:
```suggestion
title: "Bump azure-cli version"
```

File: updatecli/updatecli.d/azure-cli.yml [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+      owner: "{{ .github.owner }}"
+      repository: "{{ .github.repository }}"
+      token: "{{ requiredEnv .github.token }}"
+      username: "{{ .github.username }}"
+      branch: "{{ .github.branch }}"
+
+sources:
+  getAzCliVersion:
+    kind: githubRelease
+    name: Get the latest AzCli version
```
Comment (lemeurherve) [inline updatecli/updatecli.d/azure-cli.yml:19]:
```suggestion
    name: Get the latest azure-cli version
```

File: updatecli/updatecli.d/azure-cli.yml [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+pullrequests:
+  default:
+    kind: github
+    scmID: default
+    targets:
+      - updateCstVersion
+      - updateDockerfileArgVersion
+    spec:
+      labels:
+        - dependencies
```
Comment (lemeurherve) [inline updatecli/updatecli.d/azure-cli.yml:80]:
```suggestion
        - dependencies
        - azure-cli
```

**Review by lemeurherve — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (dduportal) [reply Dockerfile:119]: https://github.com/jenkins-infra/docker-helmfile/pull/84

File: updatecli/updatecli.d/azure-cli.yml (hunk 3 above)
Comment (dduportal) [reply updatecli/updatecli.d/azure-cli.yml:2]: https://github.com/jenkins-infra/docker-helmfile/pull/84

File: updatecli/updatecli.d/azure-cli.yml (hunk 5 above)
Comment (dduportal) [reply updatecli/updatecli.d/azure-cli.yml:80]: https://github.com/jenkins-infra/docker-helmfile/pull/84

File: updatecli/updatecli.d/azure-cli.yml (hunk 4 above)
Comment (dduportal) [reply updatecli/updatecli.d/azure-cli.yml:19]: https://github.com/jenkins-infra/docker-helmfile/pull/84

**Pull request merged**